llvm/include/llvm/CodeGen/SDPatternMatch.h

```diff
@@ -758,6 +778,40 @@ inline auto m_False() {
       m_Value()};
 }
 
+struct CondCode_match {
+  std::optional<ISD::CondCode> CCToMatch;
+  ISD::CondCode *BindCC;
+
+  explicit CondCode_match(ISD::CondCode CC) : CCToMatch(CC), BindCC(nullptr) {}
+
+  explicit CondCode_match(ISD::CondCode *CC)
+      : CCToMatch(std::nullopt), BindCC(CC) {}
```

Comment:
The CCToMatch constructor as unneeded here.

Reply:
Fixed.